src/daemon/rpmostreed-transaction-types.c

```diff
@@ -1021,10 +1014,9 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
           g_string_append_printf (txn_title, "; localinstall: %u", pkgs->len);
 
           g_ptr_array_add (pkgs, NULL);
-          if (!rpmostree_origin_add_packages (origin, (char**)pkgs->pdata, TRUE, error))
+          if (!rpmostree_origin_add_packages (origin, (char**)pkgs->pdata, TRUE,
+                                              idempotent_layering, &changed, error))
             return FALSE;
-
-          changed = TRUE;
```

Comment:
We need to do something like:
```
add_packages (..., &add_changed);
if (add_changed)
  changed = TRUE;
```

otherwise if we changed an install but not an uninstall, the latter will override right?